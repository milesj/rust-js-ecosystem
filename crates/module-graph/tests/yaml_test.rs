mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod yaml {
    use super::*;

    #[test]
    fn object() {
        let sandbox = create_sandbox("yaml");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "object.yaml"));
    }

    #[test]
    fn array() {
        let sandbox = create_sandbox("yaml");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "array.yaml"));
    }

    #[test]
    fn primitive() {
        let sandbox = create_sandbox("yaml");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "primitive.yaml"));
    }
}
