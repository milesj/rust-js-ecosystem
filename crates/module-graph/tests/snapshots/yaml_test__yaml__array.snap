---
source: crates/module-graph/tests/yaml_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"array.yaml\")"
---
{
    1: Module {
        exports: [
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Default,
                        symbol_id: None,
                        name: "default",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/array.yaml",
        query: None,
        source: YamlModule {
            data: Sequence [
                Number(1),
                Number(2),
                Number(3),
            ],
            source: "- 1\n- 2\n- 3\n",
        },
    },
}
